src/librustc_typeck/diagnostics.rs

```diff
@@ -2752,6 +2793,28 @@ fn main() {
 ```
 "##,
 
+E0230: r##"
+The trait has more type parameters specified than appear in it's definition.
+Erroneous example code:
+
+```compile_fail,E0230
+#![feature(on_unimplemented)]
+#[rustc_on_unimplemented = "Trait error on `{Self}` with `<{A},{B},{C}>`"]
+// error: there is no type parameter C on trait TraitWithThreeParams
+trait TraitWithThreeParams<A,B>
+{}
+```
+
+Include the correct number of type params and compilation should proceed:
```

Comment:
"params" -> "parameters"
"compilation" -> "the compilation" (I find it better this way but totally open to debate)